src/events.rs

```diff
@@ -281,9 +306,83 @@ mod tests {
+                                    "ExtrinsicSuccess".to_string(),
+                                ),
+                                arguments: DecodeDifferent::Decoded(vec![
+                                    "DispatchInfo".to_string()
+                                ]),
+                                documentation: DecodeDifferent::Decoded(vec![]),
+                            },
+                            EventMetadata {
+                                name: DecodeDifferent::Decoded(
+                                    "ExtrinsicFailed".to_string(),
+                                ),
+                                arguments: DecodeDifferent::Decoded(vec![
+                                    "DispatchError".to_string(),
+                                    "DispatchInfo".to_string(),
+                                ]),
+                                documentation: DecodeDifferent::Decoded(vec![]),
+                            },
+                        ])),
+                        constants: DecodeDifferent::Decoded(vec![]),
+                        errors: DecodeDifferent::Decoded(vec![
+                            ErrorMetadata {
+                                name: DecodeDifferent::Decoded(
+                                    "InvalidSpecName".to_string(),
+                                ),
+                                documentation: DecodeDifferent::Decoded(vec![]),
+                            },
+                            ErrorMetadata {
+                                name: DecodeDifferent::Decoded(
+                                    "SpecVersionNeedsToIncrease".to_string(),
+                                ),
+                                documentation: DecodeDifferent::Decoded(vec![]),
+                            },
+                            ErrorMetadata {
+                                name: DecodeDifferent::Decoded(
+                                    "FailedToExtractRuntimeVersion".to_string(),
+                                ),
+                                documentation: DecodeDifferent::Decoded(vec![]),
+                            },
+                            ErrorMetadata {
+                                name: DecodeDifferent::Decoded(
+                                    "NonDefaultComposite".to_string(),
+                                ),
+                                documentation: DecodeDifferent::Decoded(vec![]),
+                            },
+                        ]),
+                        index: 0,
+                    }]),
+                    extrinsic: ExtrinsicMetadata {
+                        version: 0,
+                        signed_extensions: vec![],
+                    },
+                }),
+            ))
+            .unwrap(),
+        );
+
+        let input = hex::decode("0c00000000000000482d7c0900000000020000000100000000010300035884723300000000000000000200000000010300035884723300000000000000").unwrap();
```

Comment:
Can you add a comment describing what this encoded value looks like in a human readable form